ServerCore/Pages/Threads/PuzzleThreads.cshtml.cs

```diff
@@ -41,21 +53,56 @@ public async Task<IActionResult> OnGetAsync()
 
             IQueryable<Message> messages = this._context.Messages.Where(message => message.Puzzle != null
                     && message.Puzzle.EventID == Event.ID);
+            this.Title = "All help threads";
 
             // Based on the event role, filter the messages further.
             if (EventRole == EventRole.play)
             {
-                Team = await GetTeamAsync();
+                Team team = await GetTeamAsync();
                 messages = messages.Where(message => 
-                    (Team != null && message.TeamID.Value == Team.ID) 
+                    (team != null && message.TeamID.Value == team.ID) 
                     || message.SenderID == LoggedInUser.ID);
             }
+            else if (puzzleId.HasValue)
+            {
+                Puzzle puzzle = this._context.Puzzles.Where(puzzle => puzzle.EventID == this.Event.ID && puzzle.ID == puzzleId.Value).FirstOrDefault();
```

Comment:
use async/await here and all other queries, via FirstOrDefaultAsync which is used in many many places in our codebase